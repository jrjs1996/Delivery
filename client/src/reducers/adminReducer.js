import { LOGIN_ADMIN, FETCH_ADMINS } from '../actions/types';

const initialState = {
  currentAdmin: {},
  admins: [],
};

export default function (state = initialState, action) {
  switch (action.type) {
    case LOGIN_ADMIN:
      return {
        ...state,
        currentAdmin: action.payload,
      };
    case FETCH_ADMINS:
      return {
        ...state,
        admins: action.payload,
      };
    default:
      return state;
  }
}
