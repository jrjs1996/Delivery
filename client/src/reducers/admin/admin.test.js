import adminReducer from './admin';
import {
  ADD_ADMIN, CREATE_ADMIN, DELETE_CURRENT_ADMIN, FETCH_ADMINS, UPDATE_ADMIN, UPDATE_CURRENT_ADMIN,
} from '../../actions/types';
import { adminsMock, createAction } from '../../tests/mocks';

const createState = (admins, currentAdmin) => ({
  admins,
  currentAdmin,
});

describe('Admin reducer', () => {
  it('Initial state is empty', () => {
    const initialState = adminReducer(undefined, 'FAKE_ACTION');
    expect(initialState).toMatchSnapshot();
  });

  it("Returns given state when an action doesn't match", () => {
    const state = adminReducer({ mockState: 'MockState' }, 'FAKE_ACTION');
    expect(state).toMatchSnapshot();
  });

  it('Inserts admin into admins when calling CREATE_ADMIN', () => {
    const initialAdmins = { [adminsMock[0]._id]: adminsMock[0] };
    const state = createState(initialAdmins, undefined);
    const action = createAction(CREATE_ADMIN, adminsMock[1]);
    expect(adminReducer(state, action)).toMatchSnapshot();
  });

  it('Removes the current admin when calling DELETE_CURRENT_ADMIN', () => {
    const state = createState(undefined, adminsMock[0]);
    const action = createAction(DELETE_CURRENT_ADMIN);
    expect(adminReducer(state, action)).toMatchSnapshot();
  });

  it('Populates admins when calling FETCH_ADMINS', () => {
    const state = createState({}, {});
    const action = createAction(FETCH_ADMINS, adminsMock);
    expect(adminReducer(state, action)).toMatchSnapshot();
  });

  it('Updates given admin in admins when calling UPDATE_ADMIN', () => {
    const initialAdmins = {
      [adminsMock[0]._id]: adminsMock[0],
      [adminsMock[1]._id]: adminsMock[1],
    };
    const state = createState(initialAdmins, undefined);
    const action = createAction(UPDATE_ADMIN, { _id: adminsMock[0]._id, username: 'UPDATED_INFO' });
    expect(adminReducer(state, action)).toMatchSnapshot();
  });

  it('Updates current admin when calling UPDATE_CURRENT_ADMIN', () => {
    const state = createState(undefined, { username: 'INIT_ADMIN' });
    const action = createAction(UPDATE_CURRENT_ADMIN, { username: 'UPDATED_INFO' });
    expect(adminReducer(state, action)).toMatchSnapshot();
  });
});
