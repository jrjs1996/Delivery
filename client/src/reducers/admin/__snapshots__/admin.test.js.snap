// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin reducer Initial state is empty 1`] = `
Object {
  "admins": Object {},
  "currentAdmin": Object {},
}
`;

exports[`Admin reducer Inserts admin into admins when calling CREATE_ADMIN 1`] = `
Object {
  "admins": Object {
    "0": Object {
      "__v": 0,
      "_id": "0",
      "username": "FirstAdmin",
    },
    "1": Object {
      "__v": 0,
      "_id": "1",
      "username": "SecondAdmin",
    },
  },
  "currentAdmin": undefined,
}
`;

exports[`Admin reducer Populates admins when calling FETCH_ADMINS 1`] = `
Object {
  "admins": Object {
    "0": Object {
      "__v": 0,
      "_id": "0",
      "username": "FirstAdmin",
    },
    "1": Object {
      "__v": 0,
      "_id": "1",
      "username": "SecondAdmin",
    },
  },
  "currentAdmin": Object {},
}
`;

exports[`Admin reducer Removes the current admin when calling DELETE_CURRENT_ADMIN 1`] = `
Object {
  "admins": undefined,
  "currentAdmin": Object {},
}
`;

exports[`Admin reducer Returns given state when an action doesn't match 1`] = `
Object {
  "mockState": "MockState",
}
`;

exports[`Admin reducer Updates current admin when calling UPDATE_CURRENT_ADMIN 1`] = `
Object {
  "admins": undefined,
  "currentAdmin": Object {
    "username": "UPDATED_INFO",
  },
}
`;

exports[`Admin reducer Updates given admin in admins when calling UPDATE_ADMIN 1`] = `
Object {
  "admins": Object {
    "0": Object {
      "__v": 0,
      "_id": "0",
      "username": "UPDATED_INFO",
    },
    "1": Object {
      "__v": 0,
      "_id": "1",
      "username": "SecondAdmin",
    },
  },
  "currentAdmin": undefined,
}
`;
